services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/links/LinkSpec.groovy

```diff
@@ -0,0 +1,87 @@
+package org.openkilda.functionaltests.spec.northbound.links
+
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.northbound.dto.links.LinkParametersDto
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.service.northbound.NorthboundService
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.client.HttpClientErrorException
+
+class LinkSpec extends BaseSpecification {
+    @Autowired
+    NorthboundService northboundService
+    @Autowired
+    TopologyDefinition topology
+
+    @Value('${discovery.timeout}')
+    int discoveryTimeout
```

Comment:
Discovery timeout is a little bit different thing. Discovery timeout takes place when there was *no port down event*, but discovery packets stopped delivering on the other side of ISL. (basically the ISL broke somewhere in the middle, but there's still physical electric signal on both ports). This value is set to 30 seconds in prod.
When bringing the port down Kilda almost immediately understands that the ISL should be brought down (ignoring the discovery packets results. port is down thus ISL should be FAILED immediately).
On line 64 it should be enough to wait for WAIT_OFFSET to give kilda time to change the ISL status (which should be done almost immediately)
On line 82 you should wait for **discoveryInterval**(not discoveryTimeout) + WAIT_OFFSET, to let Kilda receive at least 1 discovery packet for that ISL and add that ISL to the system as DISCOVERED. discoveryInterval is usually about 3-5seconds (differs for virtual/staging/prod), this means that kilda sends a discovery packet in every possible direction every 3-5 seconds.